internal/app/auth-srv.go

```diff
@@ -0,0 +1,211 @@
+package app
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/caos/oidc/pkg/op"
+	"github.com/gorilla/mux"
+	"github.com/labstack/echo/v4"
+	"github.com/reearth/reearth-backend/internal/app/appauth"
+	"github.com/reearth/reearth-backend/internal/usecase/interactor"
+	"github.com/reearth/reearth-backend/internal/usecase/interfaces"
+
+	"github.com/golang/gddo/httputil/header"
+)
+
+func authEndPoints(ctx context.Context, e *echo.Echo, r *echo.Group, cfg *ServerConfig) {
+
+	userUsecase := interactor.NewUser(cfg.Repos, cfg.Gateways, cfg.Config.SignupSecret)
+
+	domain, err := url.Parse(cfg.Config.Auth.Domain)
+	if err != nil {
+		panic("not valid auth domain")
+	}
+	domain.Path = "/"
+
+	config := &op.Config{
+		Issuer:                domain.String(),
+		CryptoKey:             sha256.Sum256([]byte(cfg.Config.Auth.Key)),
+		GrantTypeRefreshToken: true,
+	}
+	storage := appauth.NewAuthStorage(&appauth.StorageConfig{
+		Domain: domain.String(),
+		Debug:  cfg.Debug,
+	})
+	handler, err := op.NewOpenIDProvider(
+		ctx,
+		config,
+		storage,
+		op.WithHttpInterceptors(jsonToFormHandler()),
+		op.WithHttpInterceptors(setURLVarsHandler()),
+		op.WithCustomKeysEndpoint(op.NewEndpoint(".well-known/jwks.json")),
+	)
+	if err != nil {
+		e.Logger.Fatal(err)
+	}
+
+	router := handler.HttpHandler().(*mux.Router)
+
+	err = router.Walk(muxToEchoMapper(r))
+	if err != nil {
+		return
+	}
+
+	// Actual login endpoint
+	r.POST("api/login", login(ctx, cfg, storage, userUsecase))
+
+}
+
+func setURLVarsHandler() func(handler http.Handler) http.Handler {
+	return func(handler http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.URL.Path != "/authorize/callback" {
+				handler.ServeHTTP(w, r)
+				return
+			}
+
+			r2 := mux.SetURLVars(r, map[string]string{"id": r.URL.Query().Get("id")})
+			handler.ServeHTTP(w, r2)
+		})
+	}
+}
+
+func jsonToFormHandler() func(handler http.Handler) http.Handler {
+	return func(handler http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.URL.Path != "/oauth/token" {
+				handler.ServeHTTP(w, r)
+				return
+			}
+
+			if r.Header.Get("Content-Type") != "" {
+				value, _ := header.ParseValueAndParams(r.Header, "Content-Type")
+				if value != "application/json" {
+					// Content-Type header is not application/json
+					handler.ServeHTTP(w, r)
+					return
+				}
+			}
+
+			fmt.Println("JSON -> Form: " + r.URL.String())
+			err := r.ParseForm()
+			if err != nil {
+				return
+			}
+
+			var result map[string]string
+
+			err = json.NewDecoder(r.Body).Decode(&result)
+			if err != nil {
```

Comment:
```suggestion
			if err = json.NewDecoder(r.Body).Decode(&result); err != nil {
```

Minimize variable scopes